include/context.h

```diff
@@ -90,6 +91,14 @@ struct Context
         {
             m_map.erase(key);
         }
+        void set_error(const char* str, Status status = Status::InternalError)
```

Comment:
code style: do we use camel case naming or snake case naming (underscores) in public interface ? Guess we should define/select one or another, mixing both doesn't look good